src/transformStateWithClones.js

```diff
@@ -3,11 +3,38 @@
 /**
  * @param {Object} state
  * @param {Object[]} actions
- *
+ *git
  * @return {Object[]}
  */
+function deepClone(obj) {
+  return JSON.parse(JSON.stringify(obj));
+}
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = deepClone(state);
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
```

Comment:
[CODE STYLE 2]: Use switch statement if you have limited amount of conditions.

You need to change `if` to `switch` according to the checklist. And don't forget point 3 of checklist